### ISSUE TITLE
Override article could need some more explanation

### ISSUE THREAD

**Issue by NicoHaase** [brendt/stitcher.io#250]
I'm currently upgrading a first application to use PHP 8.3, and Rector recommends to use `#[Override]`. Gathering some pros and cons about this, I've found your article at https://stitcher.io/blog/override-in-php-83 that states:

> I use an IDE that prevents me from making these kinds of mistakes

Can you explain this further? How could your IDE inform you that a parent classes method was renamed and you might consider renaming the method in your own class? How could a static analyzer detect this better?

**Comment by Jacobs63** [comment]
The IDE would not really inform you that a method was renamed, however:

- If a method is overridden, I can clearly see that I'm overriding it from a parent class:
<img width="339" alt="image" src="https://github.com/brendt/stitcher.io/assets/16275303/a6affb63-2766-4b7a-8603-430e8b1c700e">
- Unlike for a method that is not overriding anything:
<img width="336" alt="image" src="https://github.com/brendt/stitcher.io/assets/16275303/3792cb94-2306-4543-b90a-0538f62fc9cf">
- If my method is being overridden by another class, I can also clearly see that:
<img width="313" alt="image" src="https://github.com/brendt/stitcher.io/assets/16275303/86bb84dd-edd5-4391-bf4f-47db30618708">
- If I'd like to rename a method, I should utilize the [Rename refactorings shortcut](https://www.jetbrains.com/help/phpstorm/rename-refactorings.html), which would automatically rename the method in all classes that override it.

In summary, my IDE prevents me making mistakes while also providing me with additional information.
